src/realm/util/encrypted_file_mapping.cpp

```diff
@@ -533,6 +533,29 @@ void EncryptedFileMapping::write_page(size_t local_page_ndx) noexcept
         m_chunk_dont_scan[chunk_ndx] = 0;
 }
 
+void EncryptedFileMapping::write_and_update_all(size_t local_page_ndx, size_t begin_offset, size_t end_offset) noexcept
```

Comment:
In non-test code this is at most one extra copy, right? Since we share mappings between SharedGroups it's just updating the single read mapping for that section of the file with the changes to the write mapping?

Reply:
That should be the case, yes.